crates/runtime/src/tools/mcp/catalog.rs

```diff
@@ -23,29 +23,38 @@ use mcp_client::{
 use mcp_core::Tool as McpTool;
 use snafu::ResultExt;
 use std::{collections::HashMap, sync::Arc, time::Duration};
-
-use tokio::sync::RwLock;
+use tokio::{
+    sync::RwLock,
+    time::{interval, MissedTickBehavior},
+};
 
 use crate::tools::{catalog::SpiceToolCatalog, SpiceModelTool};
 
 use super::{
     tool::McpToolWrapper, MCPConfig, Result, UnderlyingInitilizationSnafu, UnderlyingTransportSnafu,
 };
 
+const HEARTBEAT_INTERVAL_SECONDS: u64 = 30; // 30 seconds
+
 pub(crate) struct McpToolCatalog {
     client: Arc<RwLock<Box<dyn McpClientTrait>>>,
 
     /// User defined name & description, not from underlying MCP.
     name: String,
+    heartbeat_task: tokio::task::JoinHandle<()>,
```

Comment:
This can use the graceful CancelleableTask that was added recently in https://github.com/spiceai/spiceai/blob/trunk/crates/runtime/src/cancellable_task.rs